pandas/types/dtypes.py

```diff
@@ -244,6 +244,119 @@ def __eq__(self, other):
         if isinstance(other, compat.string_types):
             return other == self.name
 
-        return isinstance(other, DatetimeTZDtype) and \
-            self.unit == other.unit and \
-            str(self.tz) == str(other.tz)
+        return (isinstance(other, DatetimeTZDtype) and
+                self.unit == other.unit and
+                str(self.tz) == str(other.tz))
+
+
+class PeriodDtypeType(type):
+    """
+    the type of PeriodDtype, this metaclass determines subclass ability
+    """
+    pass
+
+
+class PeriodDtype(ExtensionDtype):
+    __metaclass__ = PeriodDtypeType
+    """
+    A Period duck-typed class, suitable for holding a period with freq dtype.
+
+    THIS IS NOT A REAL NUMPY DTYPE, but essentially a sub-class of np.int64.
+    """
+    type = PeriodDtypeType
+    kind = 'O'
+    str = '|O08'
+    base = np.dtype('O')
+    num = 102
+    _metadata = ['freq']
+    _match = re.compile("period\[(?P<freq>.+)\]")
+
+    def __init__(self, freq, strict=False):
```

Comment:
I think you have to do this in ``__new__`` (pretty much the same). As we need to return a singular object (and NOT a new instance everytime you do this).

```
In [7]: PeriodDtype.construct_from_string('period[D]')
Out[7]: period[D]

In [8]: p1 = PeriodDtype.construct_from_string('period[D]')

In [9]: p2 = PeriodDtype.construct_from_string('period[D]')

In [10]: p1 == p2
Out[10]: True

In [11]: p1 is p2
Out[11]: False

In [12]: from pandas.types.dtypes import DatetimeTZDtype

In [13]: t1 = DatetimeTZDtype.construct_from_string('datetime64[ns, US/Eastern]')

In [14]: t2 = DatetimeTZDtype.construct_from_string('datetime64[ns, US/Eastern]')

In [15]: t1 == t2
Out[15]: True

In [16]: t1 is t2
Out[16]: True
```